build.py

```diff
@@ -113,20 +113,71 @@ def get_distribution_name():
     except ImportError:
         return distro.codename()
 
-def get_package_filename(p):
+def get_distribution_type(distro_id = None):
+    if distro_id is None:
+        distro_id = distro.id()
+    if distro_id == 'ol': # special handling for Oracle Linux
+        return 'rhel'
+    if distro_id in ['rhel', 'ubuntu', 'suse']:
+        return distro_id
+    distro_like = distro.like().split()
+    if len(distro_like) == 0:
+        return distro_id
+    return get_distribution_type(distro_like[0])
+
+# Does not account for derivatives that follow their own version scheme
+# (ex: RHEL7-like Amazon Linux 2)
+def get_distribution_version():
+    distro_type = get_distribution_type()
+    if distro_type == 'ubuntu':
+        return '{0}.{1}'.format(distro.major_version(), distro.minor_version())
+    elif distro_type == 'suse':
+        return distro.version()
+    distro_version = distro.major_version()
+    if distro_version in ['', None]:
+        return 'rolling'
+    else:
+        return distro_version
+
+def get_package_name(p):
+    v = get_versions()[p]
+    return 'irods-externals-{0}{1}-{2}'.format(p, v['version_string'], v['consortium_build_number'])
+    
+def get_package_version(p):
     v = get_versions()[p]
+    ver_base = '1.0-'
+    ver_pkgrev = v.get('package_revision', '0')
+    ver_pkgrev_suffix1 = ''
+    ver_pkgrev_suffix2 = ''
+    if get_package_type() == 'deb':
+        ver_pkgrev_suffix1 = '~'
+        ver_pkgrev_suffix2 = get_distribution_name()
+    else:
+        dt = get_distribution_type()
+        dv = get_distribution_version()
+        ver_pkgrev_suffix2 = dv
+        if dt == 'rhel':
+            ver_pkgrev_suffix1 = '.el'
+        elif dt == 'fedora':
+            ver_pkgrev_suffix1 = '.fc'
+        else:
+            # probably suse
+            ver_pkgrev_suffix1 = '.{0}'.format(dt)
+    return '{0}{1}{2}{3}'.format(ver_base, ver_pkgrev, ver_pkgrev_suffix1, ver_pkgrev_suffix2)
+
+def get_package_filename(p):
+    n = get_package_name(p)
     a = get_package_arch()
     t = get_package_type()
-    d = ''
+    d = get_package_version(p)
```

Comment:
I'd normally push against using single letter variables, but the variables align with the function calls and this function is very short.

However, `d` doesn't appear to align with `get_package_version`. Why was `d` chosen rather than something like `pv` or `v`?

Reply:
`v` was already in use, so I just repurposed `d`. I don't mind changing it.

Reply:
Yeah, let's pick a variable name that's closer to the function name.